src/pip/_internal/exceptions.py

```diff
@@ -245,20 +260,23 @@ class HashMismatch(HashError):
             'someone may have tampered with them.')
 
     def __init__(self, allowed, gots):
+        # type: (Dict[str, List[str]], Dict[str, _Hash]) -> None
         """
         :param allowed: A dict of algorithm names pointing to lists of allowed
             hex digests
         :param gots: A dict of algorithm names pointing to hashes we
             actually got from the files under suspicion
         """
-        self.allowed = allowed
-        self.gots = gots
+        self.allowed = allowed  # type: Dict[str, List[str]]
+        self.gots = gots  # type: Dict[str, _Hash]
```

Comment:
I'd think these are redundant, as mypy can infer that from the constructor arguments.

Reply:
Agreed, and removed in https://github.com/pypa/pip/pull/8296/commits/5c24a1eaeeccd82232252090a9c14979ddb5b470

Reply:
Also @sbidoul , I apologize if this is the correct place for this, but I also wanted to know your thoughts on
 PR  #8008 w.r.t adding `json` format to `pip list`, I have made some changes there, and would appreciate if you  have some feedback on it to move it to a closure, and then proceed on the implementation PR for it.

I tried to ask you on that PR a few times, but seems like you might not have been receiving notifications from it.